packages/typespec-test/test/todo_non_branded/generated/typespec-ts/review/todo-non-branded.api.md

```diff
@@ -4,605 +4,239 @@
-    (path: "/items"): TodoItemsList;
-    (path: "/items/{id}", id: number): TodoItemsGet;
-    (path: "/items/{itemId}/attachments", itemId: number): TodoItemsAttachmentsList;
+export interface InvalidTodoItem extends ApiError {
 }
 
 // @public
-export type TodoAttachmentOutput = TodoFileAttachmentOutput | TodoUrlAttachmentOutput;
-
-// @public (undocumented)
-export type TodoClient = Client & {
-    path: Routes;
-};
-
-// @public
-export interface TodoClientOptions extends ClientOptions {
-}
-
-// @public (undocumented)
-export interface TodoFileAttachmentOutput {
-    filename: string;
-    mediaType: string;
-    todoItemId: number;
-    url: string;
-}
-
-// @public (undocumented)
-export interface TodoItem {
-    description: string;
-    id: number;
-    // (undocumented)
-    labels: TodoLabel[];
-    ownedBy: number;
-    status: "NotStarted" | "InProgress" | "Completed";
-    title: string;
-}
-
-// @public (undocumented)
-export interface TodoItemOutput {
-    readonly completedAt: string;
-    readonly createdAt: string;
-    readonly createdBy: number;
-    description: string;
-    id: number;
+export interface InvalidUserResponse extends ApiError {
     // (undocumented)
-    labels: TodoLabelOutput[];
-    ownedBy: number;
-    status: "NotStarted" | "InProgress" | "Completed";
-    title: string;
-    readonly updatedAt: string;
-}
-
-// @public (undocumented)
-export interface TodoItemPatch {
-    description?: string;
-    ownedBy?: number;
-    status?: "NotStarted" | "InProgress" | "Completed";
-    title?: string;
+    code: "invalid-user";
 }
 
 // @public
-export interface TodoItemsAttachmentsCreateFileAttachment200Response extends HttpResponse {
+export interface PageTodoAttachment {
     // (undocumented)
-    status: "200";
+    items: TodoAttachment[];
 }
 
 // @public
-export interface TodoItemsAttachmentsCreateFileAttachment404Response extends HttpResponse {
-    // (undocumented)
-    status: "404";
-}
-
-// @public (undocumented)
-export interface TodoItemsAttachmentsCreateFileAttachmentBodyParam {
-    // (undocumented)
-    body: FormData | Array<{
-        name: "contents";
-        body: string | Uint8Array | ReadableStream<Uint8Array> | NodeJS.ReadableStream | File;
-        filename?: string;
-        contentType?: string;
-    }>;
+export interface Standard4XXResponse extends ApiError {
 }
 
-// @public (undocumented)
-export interface TodoItemsAttachmentsCreateFileAttachmentMediaTypesParam {
-    // (undocumented)
-    contentType: "multipart/form-data";
-}
-
-// @public (undocumented)
-export type TodoItemsAttachmentsCreateFileAttachmentParameters = TodoItemsAttachmentsCreateFileAttachmentMediaTypesParam & TodoItemsAttachmentsCreateFileAttachmentBodyParam & RequestParameters;
-
 // @public
-export interface TodoItemsAttachmentsCreateUrlAttachment200Response extends HttpResponse {
-    // (undocumented)
-    status: "200";
+export interface Standard5XXResponse extends ApiError {
```

Comment:
Three things here and issues tracked: https://github.com/Azure/autorest.typescript/issues/2931
- please note the error model will be skipped to generate if we upgrade to the latest tcgc generated, but we should generate these customized error models
- not sure we can map core's error response to customer's ones
- there exists status code ranges which we may not support yet?

Reply:
https://github.com/microsoft/typespec/issues/5203#issuecomment-2504951309 po should be covered.